### PR TITLE
fix(dropdown-item): match padding for RTL vs LTR and links vs non-links, consistent focus and hover states for all

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #3349

## Summary

Dropdown items now have consistent padding for RTL/LTR as well as links/non-links.
Focus and Hover states are now consistent for both RTL/LTR and links/non-links.

Observe the behavior on demo Codepen before:
https://codepen.io/ellupp/pen/YzxZaQz?editors=1000

Current

https://user-images.githubusercontent.com/19231036/140199954-25bb7354-5879-4839-be77-d837c4fe329e.mov

### CODE REVIEW

File: src/components/calcite-input/calcite-input.tsx [hunk 1]
```diff
@@ -512,7 +512,6 @@ export class CalciteInput implements LabelableComponent {
     if (direction === "up" && ((!inputMax && inputMax !== 0) || inputVal < inputMax)) {
       newValue = (inputVal + inputStep).toString();
     }
-
```
Comment (jcfranco) [inline src/components/calcite-input/calcite-input.tsx:515]: Please revert this as it is not related to the fix. 

Sidebar: IMO, newlines between if statements help with readability (easier to scan vertically).

File: src/components/calcite-dropdown-item/calcite-dropdown-item.scss [hunk 2]
```diff
@@ -111,7 +119,8 @@
   }
 }
 
-:host([active]) .container:not(.container--none-selection) {
+:host([active]) .container:not(.container--none-selection),
+:host([active]) .container--link a {
```
Comment (jcfranco) [inline src/components/calcite-dropdown-item/calcite-dropdown-item.scss:123]: Can you target the anchor's class (`dropdown-link`) instead of the element? This helps avoid cases where the style may break if the element structure is updated.

This also applies to the selectors above.

File: src/components/calcite-dropdown-item/calcite-dropdown-item.scss [hunk 3]
```diff
@@ -55,13 +55,16 @@
   @apply ml-auto mr-0;
 }
 
+.calcite--rtl a.dropdown-link {
```
Comment (jcfranco) [inline src/components/calcite-dropdown-item/calcite-dropdown-item.scss:58]: It think you can make the selector less specific and remove the `a` by moving this to the `.calcite-rtl` selector below: https://github.com/Esri/calcite-components/blob/9cc831345bdf987eb14cd86ddb90bbb1d385ee2f/src/components/calcite-dropdown-item/calcite-dropdown-item.scss#L115-L120

File: src/components/calcite-dropdown-item/calcite-dropdown-item.scss [hunk 4]
```diff
@@ -122,19 +131,6 @@
   }
 }
 
-// none selection mode and scales
```
Comment (jcfranco) [inline src/components/calcite-dropdown-item/calcite-dropdown-item.scss:125]: Nice! ✨ :broom:✨

**Review by jcfranco — CHANGES_REQUESTED** [review]
This is looking great! Just had a few comments and this will be good to merge. 

Also, can you update the PR title to follow Conventional Commits?

**Review by jcfranco — APPROVED** [review]
Awesome! 🤩

Great, descriptive PR title by the way.

**Comment by asangma** [comment]
🚀
